src/molecules/lists/drag-and-drop-list-box.tsx

```diff
@@ -0,0 +1,212 @@
+                                item.label != null && (
+                                    <div
+                                        className={`${ListBoxItemClassName}__label`}>
+                                        {item.label}
+                                    </div>
+                                )}
+                                <div
+                                    className={`${ListBoxItemClassName}__text`}>
+                                    {item.text}
+                                </div>
+                                {// if
+                                props.onActionClick != null && (
+                                    <Button
+                                        type={ButtonTypes.Button}
+                                        cssClassName={`${ListBoxItemClassName}__action`}
+                                        onClick={() =>
+                                            props.onActionClick!(
+                                                // @ts-ignore it's complaining that T could be a subclass of T
+                                                item.id
+                                            )
+                                        }
+                                        size={ButtonSizes.Small}
+                                        style={ButtonStyles.TertiaryAlt}>
+                                        {props.actionText ?? "Action"}
+                                    </Button>
+                                )}
+                                {item.customAction}
+                            </div>
+                        )}
+                    </Draggable>
+                ))}
+            </React.Fragment>
+        );
+    });
+
+    const cssIsDragging = (snapshot: DroppableStateSnapshot) =>
+        snapshot.isDraggingOver ? "-dragging-container" : "";
+
+    return (
+        <DragDropContext onDragEnd={handleDragEnd}>
+            <Droppable droppableId={props.droppableId}>
+                {(
+                    provided: DroppableProvided,
+                    snapshot: DroppableStateSnapshot
+                ) => (
+                    <div
+                        {...provided.droppableProps}
+                        ref={provided.innerRef}
+                        className={`${COMPONENT_CLASS} ${cssIsDragging(
+                            snapshot
+                        )}`}>
+                        {// if
+                        hasItems && <DraggableItemList items={items} />}
+                        {provided.placeholder}
+                    </div>
+                )}
+            </Droppable>
+        </DragDropContext>
+    );
+};
+
+// #endregion Component
+
+// -------------------------------------------------------------------------------------------------
+// #region Exports
+// -------------------------------------------------------------------------------------------------
+
+export default DragAndDropListBox;

```

Comment:
Nitpick - believe we are preferring named exports for our open source libraries.